cpu/cortexm_common/include/irq_arch.h

```diff
@@ -22,27 +22,54 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include "cpu_conf.h"
+#include "kernel_defines.h"
+#include "debug_irq_disable.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief   Start SysTick timer to measure time spent with IRQ disabled
+ */
+static inline void _irq_debug_start_count(void)
+{
+    SysTick->VAL  = 0;
```

Comment:
wouldn't this work better if we keep the systick running and store the lock time 

and than `return reload_mask & ( unlock_time -locktime)`

Reply:
Why do you think that's better? We'd need an extra variable and start / stop would no longer by symmetric, we'd need an extra function to start the SysTick.

Reply:
The way this PR does it is: ether use systick for this xor for something else (e.g.: your countdowntimer)
The way would allow you to have any Initialisation on that systick (counting down or systicking around)  and do this  and all that needs is  `static uint32_t`

i am not sure if it also would be quicker since these are 3 memory write accesses and mine has 1 read and 1 write for start and this has 1 read and 1 write for stop vs mine has 2 reads.

Reply:
The Countdown timer wouldn't run the SysTick continuously either, only when there is a countdown scheduled. 

And I'm not *that* concerned about a few instructions here - the UART printouts will ensure that this is purely a debug feature and can't be used for anything performance critical :wink: